src/core/view/TextView.cpp

```diff
@@ -43,54 +44,71 @@ void TextView::updatePangoFont(PangoLayout* layout, const Text* t) {
     pango_font_description_free(desc);
 }
 
-void TextView::drawText(cairo_t* cr, const Text* t) {
-    cairo_save(cr);
+void TextView::draw(const Context& ctx) const {
+    if (text->isInEditing()) {
+        return;
+    }
 
-    cairo_translate(cr, t->getX(), t->getY());
+    cairo_save(ctx.cr);
 
-    PangoLayout* layout = initPango(cr, t);
-    string str = t->getText();
-    pango_layout_set_text(layout, str.c_str(), str.length());
+    // make elements without audio translucent when highlighting elements with audio
+    if (ctx.fadeOutNonAudio && text->getAudioFilename().empty()) {
+        cairo_set_operator(ctx.cr, CAIRO_OPERATOR_OVER);
+        Util::cairo_set_source_rgbi(ctx.cr, text->getColor(), OPACITY_NO_AUDIO);
+    } else {
+        cairo_set_operator(ctx.cr, CAIRO_OPERATOR_SOURCE);
+        Util::cairo_set_source_rgbi(ctx.cr, text->getColor());
+    }
+
+    cairo_translate(ctx.cr, text->getX(), text->getY());
 
-    pango_cairo_show_layout(cr, layout);
+    PangoLayout* layout = initPango(ctx.cr, text);
+    std::string str = text->getText();
+    pango_layout_set_text(layout, str.c_str(), static_cast<int>(str.length()));
+
+    pango_cairo_show_layout(ctx.cr, layout);
 
     g_object_unref(layout);
 
-    cairo_restore(cr);
+    cairo_restore(ctx.cr);
 }
 
-auto TextView::findText(const Text* t, string& search) -> std::vector<XojPdfRectangle> {
+auto TextView::findText(const Text* t, std::string& search) -> std::vector<XojPdfRectangle> {
+    size_t patternLength = search.length();
+    if (patternLength == 0) {
+        return {};
+    }
+
     cairo_surface_t* surface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, 1, 1);
     cairo_t* cr = cairo_create(surface);
 
     PangoLayout* layout = initPango(cr, t);
-    string str = t->getText();
-    pango_layout_set_text(layout, str.c_str(), str.length());
+    std::string str = t->getText();
```

Comment:
Please make the variable name more descriptive. Also, reuse the variable further down.